ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -830,9 +831,66 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x) (fromIntegral x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: TxSizeSplitTestSetup -> Property
+prop_TxSeq_splitAfterTxSize tss =
+      property $ txSizeSum (txTickets before) <= tssTxSizeToSplitOn
+  where
+    TxSizeSplitTestSetup { tssTxSizeToSplitOn } = tss
+
+    (before, _after) = splitAfterTxSize txseq tssTxSizeToSplitOn
+
+    txseq :: TxSeq Int
+    txseq = txSizeSplitTestSetupToTxSeq tss
+
+    txSizeSum :: [TxTicket tx] -> TxSizeInBytes
+    txSizeSum = sum . map txTicketTxSizeInBytes
+
+
+{-------------------------------------------------------------------------------
+  TxSizeSplitTestSetup
+-------------------------------------------------------------------------------}
+
+data TxSizeSplitTestSetup = TxSizeSplitTestSetup
+  { tssTxSizes         :: ![TxSizeInBytes]
+  , tssTxSizeToSplitOn :: !TxSizeInBytes
+  } deriving Show
+
+instance Arbitrary TxSizeSplitTestSetup where
+  arbitrary = do
+    let txSizeMaxBound = 10 * 1024 * 1024 -- 10MB transaction max bound
+    txSizes <- listOf $ choose (1, txSizeMaxBound)
```

Comment:
Oops. Meant to use `listOf1` after making the change specified here: https://github.com/input-output-hk/ouroboros-network/pull/1346#discussion_r358171400